app/components/autocomplete/slash_suggestion/app_command_parser/app_command_parser.ts

```diff
@@ -1093,10 +1102,14 @@ export class AppCommandParser {
         }
 
         // There have been 0 to 2 dashes in the command prior to this call, adjust.
+        const prevCharIndex = parsed.incompleteStart - 1;
         let prefix = '--';
-        for (let i = parsed.incompleteStart - 1; i > 0 && i >= parsed.incompleteStart - 2 && parsed.command[i] === '-'; i--) {
+        for (let i = prevCharIndex; i > 0 && i >= parsed.incompleteStart - 2 && parsed.command[i] === '-'; i--) {
             prefix = prefix.substring(1);
         }
+        if (prevCharIndex > 0 && parsed.command[prevCharIndex] === '—') {
```

Comment:
@larkox, maybe add a comment that the character we compare with is here an em dash (hard to identify it visually)